kratos/includes/model_part_io.h

```diff
@@ -354,8 +354,10 @@ class KRATOS_API(KRATOS_CORE) ModelPartIO : public IO
     void ReadNodalVectorialVariableData(NodesContainerType& rThisNodes, TVariableType& rVariable, TDataType Dummy);
 
     void ReadElementalDataBlock(ElementsContainerType& rThisElements);
-    
-    void WriteElementalDataBlock(ElementsContainerType& rThisElements);
+    template<class TDataType>
+    void WriteDataBlock(const TDataType& rThisData, const std::string& DataName);
```

Comment:
IMO the type and name of the arguments are very ambiguous. Usually DataType is refered to  values like double, vector, matrix, and data is a value stored in these types not a container of element or conditions.